SignalServiceKit/src/Messages/TSMessagesManager.m

```diff
@@ -494,6 +495,14 @@ - (void)handleEnvelope:(OWSSignalServiceProtosEnvelope *)envelope plaintextData:
     if (envelope.hasContent) {
         OWSSignalServiceProtosContent *content = [OWSSignalServiceProtosContent parseFromData:plaintextData];
         DDLogInfo(@"%@ handling content: <Content: %@>", self.tag, [self descriptionForContent:content]);
+        
+        if ([content hasProfileKey]) {
+            NSData *profileKey = [content profileKey];
+            NSString *recipientId = envelope.source;
+            [OWSProfilesManager.sharedManager setProfileKey:profileKey
```

Comment:
Does this need to be done on the main thread? (just thinking about our future work to move message processing off the main thread)